### PR TITLE
docs: provide guidance on using Consul DNS

### PR DESCRIPTION
**Pull request by tgross**
Add a standalone section to the Consul integration docs showing how to configure both the Consul agent and the workload to take advantage of Consul DNS. Include a reference to the new transparent proxy feature as well.

Fixes: https://github.com/hashicorp/nomad/issues/18305
Preview link: https://nomad-2z0hxy8kc-hashicorp.vercel.app/nomad/docs/integrations/consul#dns

---

I can backport this to 1.5.x-1.7.x, but I'll need to remove the reference to tproxy, so I'll do that by hand.

### CODE REVIEW

File: website/content/docs/integrations/consul/index.mdx [hunk 1]
```diff
@@ -60,6 +60,19 @@ Consul data, Vault secrets, or just general configurations within a Nomad task.
...
+# DNS
+
+To provide Consul DNS to Nomad workloads using [bridge or CNI networking
+mode][], you will need to configure Consul's DNS listener to be exposed to the
+workload network namepsace. You can do this without exposing the Consul agent on
+a public IP by setting the Consul `bind_addr` to bind on a private IP address
+(the default is to use the `client_addr`).
+
+You will also need to set the [nameserver][] to the address. This address is
+exposed as the `consul.dns.addr` node attribute. Alternately, you can use Consul
```
Comment (lgfa29) [inline website/content/docs/integrations/consul/index.mdx:72]: Not sure if I fully understand this part. 

Say my Consul agent is configured with `bind_addr = "10.0.0.2"`, then my job would need to look something like this?

```hcl
group "..." {
  network {
   mode = "bridge"
    dns {
      servers = ["10.0.0.2"]
    }
  }
}
```

Does the non-default port Consul uses for DNS affect this?

**Review by lgfa29 — APPROVED** [review]
Mostly me being confused about DNS (it's always DNS 😅)

File: website/content/docs/integrations/consul/index.mdx (hunk 1 above)
Comment (tgross) [reply website/content/docs/integrations/consul/index.mdx:72]: Oh, that's a good point. Lemme double-check if you can give the nameserver a port and have that work (it'd have to be set to `consul.dns.port` in that case), but if not, then you need to bind Consul DNS to port 53 too.

Comment (tgross) [reply website/content/docs/integrations/consul/index.mdx]: Nope, that doesn't work... I've fixed this section up and added a link to https://developer.hashicorp.com/consul/tutorials/networking/dns-forwarding which has a lot more detail.

File: website/content/docs/integrations/consul/index.mdx [hunk 2]
```diff
@@ -174,3 +194,7 @@ Consul, with some exceptions.
 [int_consul_mesh]: /nomad/docs/integrations/consul/service-mesh
 [nomad_config_consul]: /nomad/docs/configuration/consul
 [service]: /nomad/docs/job-specification/service "Nomad service Job Specification"
+[bridge or CNI networking mode]: /nomad/docs/job-specification/network#mode
+[nameserver]: /nomad/docs/job-specification/network#servers
+[transparent proxy]: /nomad/docs/job-specification/transparent_proxy
+ [Forward DNS for Consul service discovery](https://developer.hashicorp.com/consul/tutorials/networking/dns-forwarding)
```
Comment (lgfa29) [inline website/content/docs/integrations/consul/index.mdx:200]: I think this extra space is tripping up the Markdown rendering 😅 
https://nomad-git-docs-consul-dns-hashicorp.vercel.app/nomad/docs/integrations/consul#dns

Comment (tgross) [reply website/content/docs/integrations/consul/index.mdx:200]: Crap, thanks. Will fix.

Comment (tgross) [reply website/content/docs/integrations/consul/index.mdx:200]: https://github.com/hashicorp/nomad/pull/20387

**Pull request merged**